diff -ur iscsitarget-0.4.15.orig/etc/initd/initd.redhat iscsitarget-0.4.15/etc/initd/initd.redhat
--- iscsitarget-0.4.15.orig/etc/initd/initd.redhat	2008-01-25 19:54:08.000000000 +0100
+++ iscsitarget-0.4.15/etc/initd/initd.redhat	2008-01-25 19:54:29.000000000 +0100
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # chkconfig: - 39 35
 # description: Starts and stops the iSCSI target
@@ -6,6 +6,14 @@
 # pidfile: /var/run/ietd.pid
 # config:  /etc/ietd.conf
 
+### BEGIN INIT INFO
+# Provides: iscsi-target
+# Required-Start: $local_fs $network $remote_fs $syslog
+# Required-Stop: $local_fs $network $remote_fs $syslog
+# Short-Description: start and stop ietd
+# Description: Starts and stops the iSCSI target
+### END INIT INFO
+
 # Source function library.
 if [ -f /etc/init.d/functions ] ; then
   . /etc/init.d/functions
@@ -26,6 +34,7 @@
 	modprobe iscsi_trgt
 	daemon /usr/sbin/ietd
 	RETVAL=$?
+	[ $RETVAL -eq 0 ] && touch /var/lock/subsys/iscsi-target
 	echo
 	return $RETVAL
 }
@@ -40,6 +49,7 @@
 	modprobe -r crc32c 2>/dev/null
 	if [ $RETVAL -eq 0 ]; then
 		echo_success
+		rm -f /var/lock/subsys/iscsi-target
 	else
 		echo_failure
 	fi
@@ -61,6 +71,16 @@
 	fi
 }
 
+try-restart()
+{
+	condrestart
+}
+
+force-reload()
+{
+	try-restart
+}
+
 status()
 {
 	PID=`pidofproc ietd`
@@ -89,7 +109,7 @@
         status
         ;;
   *)
-        echo $"Usage: $0 {start|stop|restart|status}"
+        echo $"Usage: $0 {start|stop|restart|try-restart|force-reload|status}"
         exit 1
 esac
 
